kvision-tools/kvision-gradle-plugin/src/main/kotlin/io/kvision/gradle/KVisionPlugin.kt

```diff
@@ -0,0 +1,369 @@
+    }
+
+
+    private fun KVPluginContext.registerGeneratePotFileTask(configuration: KVGeneratePotTask.() -> Unit = {}) {
+        tasks.register<KVGeneratePotTask>("generatePotFile") {
+            enabled = kvExtension.enableGradleTasks.get()
+
+            nodeJsBinary.set(kvExtension.nodeBinaryPath)
+
+            getTextExtractBin.set(
+                rootNodeModulesDir.file("gettext-extract/bin/gettext-extract")
+            )
+            configuration()
+        }
+    }
+
+
+    private fun KVPluginContext.registerConvertPoToJsonTask(
+        configuration: KVConvertPoTask.() -> Unit = {}
+    ): TaskProvider<KVConvertPoTask> {
+        return tasks.register<KVConvertPoTask>("convertPoToJsonTask") {
+            group = KVISION_TASK_GROUP
+
+            enabled = kvExtension.enableGradleTasks.get()
+
+            dependsOn(tasks.compileKotlinJs)
+
+            po2jsonBinDir.set(
+                rootNodeModulesDir.file("@rjaros/gettext.js/bin/po2json")
+            )
+
+            nodeJsBinary.set(nodeJsBinaryProvider())
+
+            configuration()
+        }
+    }
+
+
+    private fun KVPluginContext.registerZipTask(configuration: Zip.() -> Unit = {}) {
+        val webDir = layout.projectDirectory.dir("src/main/web")
+
+        tasks.register<Zip>("zip") {
+            group = PACKAGE_TASK_GROUP
+            description = "Builds ZIP archive with the application"
+
+            enabled = kvExtension.enableGradleTasks.get()
+
+            dependsOn(tasks.browserProductionWebpack)
+
+            destinationDirectory.set(layout.buildDirectory.dir("libs"))
+            from(tasks.browserProductionWebpack.map { it.destinationDirectory }) {
+                include("*.*")
+            }
+            from(webDir)
+
+            duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+            configuration()
+        }
+    }
+
+
+    /** Requires Kotlin MPP project */
+    private fun KVPluginContext.registerWorkerBundleTask() {
+        tasks.register<KVWorkerBundleTask>("workerBundle") {
+            dependsOn(tasks.workerBrowserProductionWebpack)
+
+            enabled = kvExtension.enableWorkerTasks.get()
+
+            nodeJsBin.set(nodeJsBinaryProvider())
+            webpackJs.set(
+                rootNodeModulesDir.file("webpack/bin/webpack.js")
+            )
+            webpackConfigJs.set(
+                rootProject.layout.buildDirectory.file("js/packages/${rootProject.name}-worker/webpack.config.js")
+            )
+            workerMainSrcDir.set(
+                layout.projectDirectory.dir("src/workerMain/kotlin")
+            )
+            workerJsFile.set(
+                layout.buildDirectory.file("processedResources/frontend/main/worker.js")
+            )
+
+            inputs.dir(workerMainSrcDir)
+            outputs.file(workerJsFile)
+
+            executable(nodeJsBin)
+            workingDir(
+                rootProject.layout.buildDirectory.dir("js/packages/${rootProject.name}-worker")
+            )
```

Comment:
Instead of hardcoding the directories, it would be better to use the tasks that produce these files.

E.g.

```kotlin
workerJsFile.set(
  tasks.someTaskThatCreatesWorkerJs.map { it.workerJsOutputFile } 
  // layout.buildDirectory.file("processedResources/frontend/main/worker.js")
)
```

This helps Gradle optimise the build, to re-use work and avoid repetition. 

Do you know what tasks produce or define these files?

Reply:
Unfortunately I don't. But this task is used only with a dedicated configuration I've created for webworkers - see demo project here: https://github.com/rjaros/template-webworker